acceptance-tests/src/main/java/io/blueocean/ath/SmartWebElement.java

```diff
@@ -0,0 +1,284 @@
+        WebElement el = elements.iterator().next();
+        return (T)((JavascriptExecutor)getDriver()).executeScript(script, el, elements);
+    }
+
+    /**
+     * Send an event to the matched elements - e.g. 'blur' or 'change'
+     * @param type
+     */
+    public void sendEvent(String type) {
+        forEach(e -> sendEvent(e, type));
+    }
+
+    protected void sendEvent(WebElement el, String type) {
+        StringBuilder script = new StringBuilder(
+            "return (function(a,b,c,d){" +
+                "c=document," +
+                "c.createEvent" +
+                "?(d=c.createEvent('HTMLEvents'),d.initEvent(b,!0,!0),a.dispatchEvent(d))" +
+                ":(d=c.createEventObject(),a.fireEvent('on'+b,d))})"
+        );
+        script.append("(arguments[0],'").append(type.replace("'", "\\'")).append("');");
+        ((JavascriptExecutor)getDriver()).executeScript(script.toString(), el);
+    }
+
+    protected void sendInputEvent(WebElement el) {
+        sendEvent(el, "input");
+    }
+
+    private static void validateTextElement(WebElement element) {
+        String tagName = element.getTagName().toLowerCase();
+        Preconditions.checkArgument(
+            "input".equals(tagName) || "textarea".equals(tagName),
+            "element must should be input or textarea but was %s",
+            tagName
+        );
+    }
+
+    /**
+     * Sets the matched inputs to the given text, if setting to empty string
+     * there is some special handling to clear the input such that events are
+     * properly handled across platforms by sending an additional 'oninput' event
+     * @param text text to use
+     */
+    public void setText(CharSequence... text) {
+        forEach(e -> {
+            validateTextElement(e);
+            e.clear();
+            e.sendKeys(text);
+            // If setting the text empty,
+            if (text.length == 1 && "".equals(text[0])) {
+                // b'cuz React, see: https://github.com/facebook/react/issues/8004
+                sendInputEvent(e);
+            }
+        });
+    }
+
+    @Override
+    public void clear() {
+        forEach(e -> {
+            e.clear();
+            // b'cuz React, see: https://github.com/facebook/react/issues/8004
+            sendInputEvent(e);
+        });
+    }
+
+    @Override
+    public String getTagName() {
+        WebElement el = getElement();
+        if (el == null) {
+            return null;
+        }
+        return el.getTagName();
+    }
+
+    @Override
+    public String getAttribute(String s) {
+        WebElement el = getElement();
+        if (el == null) {
+            return null;
+        }
+        return el.getAttribute(s);
+    }
+
+    @Override
+    public boolean isSelected() {
+        WebElement el = getElement();
+        if (el == null) {
+            return false;
+        }
+        return el.isSelected();
+    }
+
+    @Override
+    public boolean isEnabled() {
```

Comment:
Would this style of method implicitly wait for the element to be enabled, or do an immediate check for whether it's enabled? If the latter, it seems like it could be subject to timing failures. Is there logic we can borrow from `ExpectedConditions` here? I honestly don't know the answer but the stuff in `WaitUtil` generally seems to wait the right way, although maybe the API can be improved here?

Reply:
just checks for it being enabled. sometimes you might want to check for things enabled/disabled based on user input. could add more methods as needed e.g. `waitUntilEnabled`